automation/jenkins-hostapp-extension.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash 
+
+set -e
+
+[ $# -lt 2 ] && echo "Hostapp extension packages need to be defined" && exit 1
+[ -z "${BALENAOS_STAGING_TOKEN}" ] && echo "Balena staging token is required" && exit 1
+[ -z "${BALENAOS_PRODUCTION_TOKEN}" ] && echo "Balena production token is required" && exit 1
+[ -z "${HOSTEXT_NAME}" ] && echo "Hostapp extension name is required" && exit 1
+[ -z "${deployTo}" ] && echo "Deployment environment target is required" && exit 1
+[ -z "${VERSION_HOSTOS}" ] && echo "Balena OS version is required" && exit 1
+[ -z "${MACHINE}" ] && echo "Device type is required" && exit 1
+[ -z "${PACKAGE_TYPE}" ] && echo "Package type is required" && exit 1
+
+PACKAGES=""
+for pkg in $@; do
+	if [ -z "${PACKAGES}" ]; then
+		PACKAGES="${pkg}"
+	else
+		PACKAGES="${PACKAGES} ${pkg}"
+	fi
+done
+
+finish() {
+	result=$?
+	exit ${result}
+}
+trap finish EXIT ERR
+
+deploy_to_balena() {
+	local _exported_image_path=$1
+	local DEVELOPMENT_IMAGE=no
+	if [ "${buildFlavor}" = "dev" ]; then
+		DEVELOPMENT_IMAGE=yes
+	fi
```

Comment:
DEVELOPMENT_IMAGE can be removed if we think we'd never build .dev images

Reply:
Well, the hostapp extension is built against a package feed that is build from an image, and that image currently can be development or production. So in theory nothing prevents the hostapp extension to have content that comes from a development image. That's why I choose to leave it.
What we should do is merge production/development images so we can get rid of these complexities that move all the way up the stack.